### ISSUE TITLE
Setting bind address for domain mode

### ISSUE THREAD

**Issue by dnataraj** [bdwyertech/chef-wildfly#21]
I know this cookbook does not support standalone explicitily, but I am able to use the node.override['wildfly']['mode'] = 'domain' attribute to start up correctly in domain mode. I have to provide my own domain.xml though (since the server groups are not there). However, I am not able to access the management port from outside the VM (9990) in domain mode - only standalone. It seems like the attribute default['wildfly']['int']['mgmt']['bind'] = '0.0.0.0' only applies to standalone configurations? Is this a valid issue?

**Comment by dnataraj** [comment]
Also, it seems that overriding attributes is just not working (unless I am doing something completely wrong). In my recipe I have
node.override['wildfly']['java_opts']['xmx'] = '1024m'
node.override['wildfly']['java_opts']['xms'] = '512m'
node.override['wildfly']['java_opts']['other'] = %w(-Djboss.bind.address.management=0.0.0.0)

These just seem to have no effect.

**Comment by bdwyertech** [comment]
I haven't gotten around to making this cookbook work for domain mode, mainly because I have no experience running in domain mode, and figured others would be simply deploy their own domain.xml, as it seems you have done. If you look at the recipe `install.rb`, everything is more or less hardcoded for standalone mode.

The reason the overrides are not working for domain mode are because there is no provisioner in install.rb for domain.conf, the domain mode launch configuration.  I'll go ahead and put that in there now, but set it to only deploy domain.conf if specifically set to domain mode.

Finally, the reason the management bind address for domain mode is not working is because that should be defined in your domain.xml; there is not a separate configuration file for the interface bindings and such.  You could probably include the following in your domain.xml file to handle this utilizing this cookbook.

```ruby
    <interfaces>
        <interface name="management">
          <% if "0.0.0.0" == node['wildfly']['int']['mgmt']['bind'] %>
            <any-address/>
          <% else %>
            <inet-address value="${jboss.bind.address.management:<%= node['wildfly']['int']['mgmt']['bind'] %>}"/>
          <% end %>
        </interface>
        <interface name="public">
          <% if "0.0.0.0" == node['wildfly']['int']['pub']['bind'] %>
            <any-address/>
          <% else %>
            <inet-address value="${jboss.bind.address:<%= node['wildfly']['int']['pub']['bind'] %>}"/>
          <% end %>
        </interface>
        <!-- TODO - only show this if the jacorb subsystem is added  -->
        <interface name="unsecure">
            <!--
              ~  Used for IIOP sockets in the standard configuration.
              ~                  To secure JacORB you need to setup SSL 
              -->
            <inet-address value="${jboss.bind.address.unsecure:127.0.0.1}"/>
        </interface>
    </interfaces>
```

**Comment by bdwyertech** [comment]
The `domain.conf` overrides issue should be addressed by 48a8b9e03df58e7f1e16badb78ace640f4295250
